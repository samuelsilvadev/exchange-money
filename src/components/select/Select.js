import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import ArrowDown from '../icons/ArrowDown';

import styles from './Select.module.css';

function Select(props) {
	const { children, id, className, label, onChange, onBlur } = props;

	return (
		<div
			className={classnames(styles.selectWrapper, {
				[className]: className,
			})}
		>
			<label className={styles.selectLabel} htmlFor={id}>
				{label}
			</label>
			<div className={styles.selectIconWrapper}>
				<select
					className={styles.select}
					id={id}
					onChange={onChange}
					onBlur={onBlur}
				>
					{children}
				</select>
				<ArrowDown className={styles.selectIcon} />
			</div>
		</div>
	);
}

Select.propTypes = {
	id: PropTypes.string.isRequired,
	label: PropTypes.string.isRequired,
	className: PropTypes.string,
	children: PropTypes.node,
	onChange: PropTypes.func,
	onBlur: PropTypes.func,
};

export default Select;
