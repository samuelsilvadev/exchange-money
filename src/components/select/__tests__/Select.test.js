import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Select from '../Select';

describe('<Select />', () => {
	it('should render correctly', () => {
		const { container, getByLabelText, getByText } = render(
			<Select id="color" label="color">
				<option>Red</option>
				<option>Blue</option>
				<option>Green</option>
			</Select>
		);

		expect(getByLabelText('color')).toBeVisible();
		expect(getByText('Red')).toBeInTheDocument();
		expect(getByText('Blue')).toBeInTheDocument();
		expect(getByText('Green')).toBeInTheDocument();
		expect(container.firstChild).toMatchSnapshot();
	});

	it('should call `onChange` correctly', () => {
		const handleChange = jest.fn();

		const { getByLabelText } = render(
			<Select id="color" label="color" onChange={handleChange}>
				<option>Red</option>
				<option>Blue</option>
				<option>Green</option>
			</Select>
		);

		const select = getByLabelText('color');

		fireEvent.change(select);

		expect(handleChange).toHaveBeenCalledTimes(1);
	});
});
