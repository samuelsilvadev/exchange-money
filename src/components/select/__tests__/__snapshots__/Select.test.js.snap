// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render correctly 1`] = `
<div
  class="selectWrapper"
>
  <label
    class="selectLabel"
    for="color"
  >
    color
  </label>
  <div
    class="selectIconWrapper"
  >
    <select
      class="select"
      id="color"
    >
      <option>
        Red
      </option>
      <option>
        Blue
      </option>
      <option>
        Green
      </option>
    </select>
    <svg
      aria-hidden="true"
      class="selectIcon"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 15l-4.243-4.243 1.415-1.414L12 12.172l2.828-2.829 1.415 1.414z"
      />
    </svg>
  </div>
</div>
`;
